src/pfe/file-watcher/server/src/controllers/projectStatusController.ts

```diff
@@ -481,7 +481,7 @@ function pingInTransitApplications(): void {
                                         newState = (stateInfo.isAppUp && projectInfo.sentProjectInfo) ? AppState.started : oldState;
                                     } else if (oldState === AppState.stopping && !stateInfo.isAppUp) {
                                         newState = AppState.stopped;
-                                    } else if ( oldState === AppState.starting ) {
+                                    } else if ( oldState === AppState.starting && pingCountLimit != -1 ) {
```

Comment:
I think the design was to set the ping port number to -1 to disable the ping completely instead of ping timeout period.